frontend/src/components/ViewActivities/index.js

```diff
@@ -1,16 +1,89 @@
 import React from 'react';
+import app from '../Firebase';
+import { getUserUid } from '../AuthUtils';
+import ActivityList from './activitylist.js';
+
+import * as DB from '../../constants/database.js';
 
 /**
- * ViewActivities component.
+ * The view activities page. First checks that the current user is authorized to
+ * view the current trip (i.e. they are a collaborator for it). If so, the
+ * ActivityList component is displayed as normal. If not, an error is displayed
+ * instead.
+ *
+ * @param {Object} props This component expects the following props:
+ * - `tripId` {string} The trip's ID. This is sent to the component through the URL.
  */
 class ViewActivities extends React.Component {
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+    this.tripId = props.match.params.tripId;
+    this.state = {
+      collaborators: undefined,
+      isLoading: true,
+      error: undefined
+    }
+  }
+
+  /** @inheritdoc */
+  componentDidMount() {
+    app.firestore()
+        .collection(DB.COLLECTION_TRIPS)
+        .doc(this.tripId)
+        .get()
+        .then(doc => {
+          this.setState({
+            collaborators: doc.get(DB.TRIPS_COLLABORATORS),
+            isLoading: false,
+            error: undefined
+          });
+        })
+        .catch(e => {
+          this.setState({
+            collaborators: undefined,
+            isLoading: true,
+            error: e
+          })
+        });
+  }
+
+  /** @inheritdoc */
   render() {
-    return (
-      <div>
-        <h1>View Activities</h1>
-      </div>
-    );
+    // Case where there was a Firebase error.
+    if (this.state.error !== undefined) {
+      // TODO (Issue #74): Redirect to an error page instead.
+      return (
+        <div>
+          Oops, looks like something went wrong. Please wait a few minutes and
+          try again.
+        </div>
+      );
+    }
+    // Case where the trip details are still being fetched.
+    if (this.state.isLoading) {
+      // TODO (Issue #25): Please remember to make this a blank div in the
+      // deployed build lol.
+      return <div>Loading Part 2: Electric Boogaloo</div>;
+    }
+    // Case where the trip could not be found.
+    else if (this.state.collaborators === undefined) {
+      // TODO (Issue #74): Redirect to an error page instead.
+      return <div>Sorry, we couldn't find the trip you were looking for.</div>;
+    }
```

Comment:
I am pretty confident that this case should never happen given that `isLoading` is false from your logic above (you were able to grab the document). This is because all error handling is done on the input side. So when grabbing a doc, the collaborators field should always exist and be non-null. In the future I am looking to implement the extra firebase security rules (#47) that will absolutely ensure this but there should be no possible way that this field will not exist with my current implementation.

Reply:
Ah this check isn't for that specific case. This is when someone tries to access a tripId that doesn't exist in the Datastore, e.g. if the URL said something like `.com/view-activities/random-trip-that-doesnt-exist`. Attempting to get a field from a nonexistent document would return undefined, so when this occurs I have it output that the trip couldn't be found since it doesn't exist.

It probably _does_ look weird that I'm checking for this by looking to see if the collaborators was undefined, so I can make put a small comment explaining this if you think that should be good.

Reply:
Ahhh I see that makes sense. Yeah I think a comment wouldn't hurt.

Reply:
Sounds good, will make a comment.